### PR TITLE
network: add handler to passthrough connections

### PR DESCRIPTION
**Pull request by thc202**
Handle CONNECT requests to possibly passthrough the data received/sent.

### CODE REVIEW

**Comment by thc202** [comment]
Some tests failed in one of the builds.

**Comment by thc202** [comment]
Addressed the issue in the tests.

File: addOns/network/src/main/java/org/zaproxy/addon/network/internal/handlers/PassthroughHandler.java [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+ * Handles HTTTP CONNECT requests to passthrough the data to the server.
+ *
+ * <p>The whole pipeline is reworked to passthrough the data, otherwise the handler just removes
+ * itself.
+ */
+public class PassthroughHandler extends SimpleChannelInboundHandler<HttpMessage> {
+
+    private static final Logger LOGGER = LogManager.getLogger(PassthroughHandler.class);
+
+    private static final int CONNECT_TIMEOUT_MILLIS = (int) TimeUnit.SECONDS.toMillis(20);
```
Comment (psiinon) [inline addOns/network/src/main/java/org/zaproxy/addon/network/internal/handlers/PassthroughHandler.java:52]: Would it be useful if these were configurable?

**Review by psiinon — APPROVED** [review]

File: addOns/network/src/main/java/org/zaproxy/addon/network/internal/handlers/PassthroughHandler.java (hunk 1 above)
Comment (thc202) [reply addOns/network/src/main/java/org/zaproxy/addon/network/internal/handlers/PassthroughHandler.java:52]: I thought about that but should the user need to change it? These will run in the background.

Comment (thc202) [reply addOns/network/src/main/java/org/zaproxy/addon/network/internal/handlers/PassthroughHandler.java:52]: OK.

**Review by kingthorin — APPROVED** [review]

**Pull request merged**